site/en/blog/crx-scripting-api/index.md

```diff
@@ -0,0 +1,209 @@
+greeted the user by name when the user clicks the extension's action button (icon in the toolbar).
+In Manifest V2, we could dynamically construct a code string and execute that script in the current
+page.
+
+```js
+// Manifest V2 extension
+let givenName = "Default";
+chrome.browserAction.onClicked.addListener((tab) => {
+  chrome.tabs.executeScript({
+    code: `alert(\`Hello, ${givenName}\`)`
+  });
+});
+```
+
+While Manifest V3 extensions can't use code that isn't bundled with the extension, our goal was to
+preserve some of the dynamism that arbitrary code blocks enabled for Manifest V2 extensions. The
+function and arguments approach makes it possible for Chrome Web Store reviewers, users, and other
+interested parties to more accurately assess the risks an extension poses while also allowing
+developers to modify an extension's runtime behavior based on user settings or application state.
+
+```js
+// Manifest V3 extension
+let givenName = "Default";
+
+function greetUser(name) {
+  alert(`Hello, ${name}`);
+}
+chrome.action.onClicked.addListener((tab) => {
+  chrome.tabs.executeScript({
+    target: { tabId: tab.id },
+    func: greetUser,
+    args: [givenName],
+  });
+});
+```
+
+We also wanted to make frames more of a first-class citizen of the API. The Manifest V2 version of
+`executeScript` allowed developers to target either all frames in a tab or a specific frame in the
+tab. In Manifest V3, we replaced the optional `frame` string property in the options object with an
+optional `frames` array of strings.
+
+
+```js
+// Manifest V2 extension
+chrome.action.onClicked.addListener((tab) => {
+  chrome.scripting.executeScript({
+    target: { tabId: tab.id, frame: targetFrame1 },
+    files: ["content-script.js"],
+  });
+  chrome.scripting.executeScript({
+    target: { tabId: tab.id, frame: targetFrame2 },
+    files: ["content-script.js"],
+  });
+});
+```
+
+```js
+// Manifest V3 extension
+chrome.action.onClicked.addListener((tab) => {
+  chrome.scripting.executeScript({
+    target: { tabId: tab.id, frames: [targetFrame1, targetFrame2] },
+    files: ["content-script.js"],
+  });
+});
+```
+
+We've also improved the way we return script injection results in Manifest V3. In Manifest V2,
+`executeScript` would return an array of results. Unfortunately, this implementation didn't provide
+enough information about which frame a given result applied to,  which meant that if an error occurred when injecting into `allFrames: true` it was very difficult to track down which frame had the error.
+
+In Manifest V3, we've added more details to the result objects returned in order to clearly
+identify the tab and frame that the injection occurred. This makes it much easier for developers to
+identify and react to situations where injection fails unexpectedly.
+
+Moving forward, we're also considering how extensions can interact with installed PWAs and other
+contexts that don't conceptually map to "tabs."
+
```

Comment:
I think we need some kind of an outro here - right now, it's a very abrupt ending

Reply:
Currently drafting it locally. Next commit will flesh this out.